### ISSUE TITLE
Support passing a model instance instead of a string in relations

### ISSUE THREAD

**Issue by JDaance** [leafo/lapis#752]
![image](https://user-images.githubusercontent.com/1148112/156342331-5d9381d2-2635-449c-965b-ad3495ce03c8.png)

"Users" here have to be a string, that will be required like this

![image](https://user-images.githubusercontent.com/1148112/156342553-d2e9ead6-2a86-4e72-89e8-4db1e596dcc3.png)

I would like to be able to pass a model instance like this instead when needed:

![image](https://user-images.githubusercontent.com/1148112/156342602-a3235638-8639-4a83-8ea5-60354b74af6e.png)

**Comment by leafo** [comment]
So the reason why it exists the way it currently does it to avoid issues with circular dependencies. You may not have access to the value of the model yet. (eg. they both have a relation to each other)

Maybe an alternative general purpose solution is to allow using a function in place of the string to lazily load the value of the model instance.

**Comment by JDaance** [comment]
Yes I suspected that it might be due to circular deps! 

I would be fine with a function based approach, but I would like even better if it accepted string (current), instance (to use when no circular deps are present) and function. But I can see that the circular situation is very common, declaring the relationship on both sides.

I did a workaround by passing a string as it wants and then override get_relation_model on that specific model (matching on the string).
